src/main.rs

```diff
@@ -1,110 +1,24 @@
-use my_project::interpreter::Interpreter;
-
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    println!("Hello, world!");
-
-    let mut interpreter = Interpreter::new();
-    interpreter.run("var x 10 print x")?;
+use std::{env, fs};
 
-    let mut interpreter1 = Interpreter::new();
-    interpreter1.run("var x 10 if x == 10 print x")?; // var x 10 if x 10 var y 20 else var y 30 end print y
+use my_project::interpreter::Interpreter;
 
-    let mut interpreter2 = Interpreter::new();
-    interpreter2.run("var x 30 var y 10 add x y print x end")?;
+static ERROR_MISSING_INPUT_FILE: &str = "Missing input file";
+static ERROR_FILE_READ: &str = "Could not read file";
+static ERROR_BAD_EXTENSION: &str = "Source files must have the .nek file extension";
 
-    let mut interpreter3 = Interpreter::new();
-    let source = "var x 30 
-                  var y 10 
-                  var z 40 
-                  add x y 
-                  print x
-                  add x z
-                  print x
-                  end";
-    interpreter3.run(source)?;
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = env::args().collect();
 
-    let mut interpreter4 = Interpreter::new();
-    let source1 = "
-        var x 0
-        var y 1
-        loop x < 5
-            print x
-            add x y
-            end
-        end
-    ";
-    interpreter4.run(source1)?;
+    let source_file = args.iter().skip(1).next().expect(ERROR_MISSING_INPUT_FILE);
```

Comment:
As in main we have std::error::Error why not use this feature fully? There is also custom errors. I this case I would use `ok_or` for example. It's generally better to use ok_or instead of expect for error handling because expect will panic if the Option is None, while ok_or will convert the Option into a Result that can be handled with a question mark (?) operator or a match statement.

Reply:
Indeed, it's probably better to handle potential errors with question marks. Thank you for the feedback